spidermon/contrib/scrapy/monitors.py

```diff
@@ -311,12 +329,45 @@ def test_check_unwanted_http_codes(self):
                 code: errors_max_count for code in unwanted_http_codes
             }
 
+        requests = self.stats.get("downloader/request_count", 0)
         for code, max_errors in unwanted_http_codes.items():
             code = int(code)
             count = self.stats.get(f"downloader/response_status_count/{code}", 0)
+
+            percentage_trigger = False
+
+            if isinstance(max_errors, dict):
+                absolute_max_errors = max_errors.get("max_count")
+                percentual_max_errors = max_errors.get("max_percentage")
+
+                if not absolute_max_errors and not percentual_max_errors:
+                    max_errors = self.DEFAULT_UNWANTED_HTTP_CODES_MAX_COUNT
+
+                else:
+                    calculated_percentage_errors = int(
+                        percentual_max_errors * requests
+                        if percentual_max_errors
+                        else requests
+                    )
+
+                    max_errors = min(
+                        absolute_max_errors if absolute_max_errors else requests,
+                        calculated_percentage_errors,
+                    )
+
+                    percentage_trigger = max_errors == calculated_percentage_errors
+
+            stat_message = (
+                "This exceeds the limit of {} ({}% of {} total requests)".format(
```

Comment:
`{}%` will render as 0.5% instead of 50% for 0.5. See e.g. https://stackoverflow.com/a/23764798/939364

Also, assuming I did not get it wrong, maybe you can update tests so that they would have caught this issue.

Reply:
Hmm. I don't think it will because I added a `* 100` after it. I updated a couple cases just to be safe!